src/clr-angular/forms/common/providers/control-class.service.ts

```diff
@@ -20,4 +20,15 @@ export class ControlClassService {
     }
     return controlClasses.join(' ');
   }
+
+  // We want to remove the column classes from the input up to the container
+  initControlClass(renderer, element: HTMLElement) {
+    this.className = element.className;
+    const klasses = element.className.split(' ');
```

Comment:
This breaks when rendering server-side. Immediate fix would be to guard it, but that means you'll get the wrong classes when rendering on the server side.

This is all very static, can't you just have an `@Input() @HostBinding() class` on your control components? That should let you do the same thing without accessing native element properties.

Reply:
I've been trying to prove you right about SSR, but everything I've tried builds just fine.

Using `@Input() @HostBinding() class;` clobbers the `host: { '[class.clr-input']: 'true'}` declaration, so I'm using the element directly. However, to assuage concerns about SSR I'll add a check so it doesn't throw undefined errors.